### DISCUSSION TITLE
Single sign on

### DISCUSSION THREAD

**Discussion by wastewater-it** [cisagov/LME#510] [FAQ] [answered]
As part of CISA's Secure by Design guidance, features like SSO, MFA, and audit logging are recommended for all applications out of the box. Kibana has locked those features behind Elastic's Platinum licensing - does the upcoming LME Frontend plan to address any of these shortcomings, or have an OpenSearch install option to natively support these features?

**Comment by aarz-snl** [comment, marked as answer]
the LME Frontend will be an application that can serve tools used by LME like Sysmon, agents, etc to make it easy to install on endpoints. 

You're correct that SSO/MFA is a paid feature so it only available if users pay. Opensearch isn't designed for Elastic Agents so that wouldn't be an option. 

As of now if that was an absolute requirement for an organization to implement they would need to do this with their own proxy/SSO providers where they can put something in front of their applications.

**Comment by aarz-snl** [comment]
Something we could do longer term is provide optional modular additions like nginx and keycloak. Pending available resources. 

run a local keycloak instance to provide MFA capabilities and NGINX to handle routing the user before/after authentication
